File: 프로그래머스/lv1/42748. K번째수/K번째수.py
def solution(array, commands):
    answer = []
    
    for cmd in commands:
        temp = array[cmd[0]-1:cmd[1]]
        # print(temp)
        temp.sort()
        # print(temp)
        # print(cmd[2])
        answer.append(temp[cmd[2] - 1])
        
    
    return answer

File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.*;

class Solution {
    public static void main(String[] args) {
        Solution solution = new Solution();
    }
    
    public List<Integer> solution(int[] array, int[][] commands) {
        List<Integer> arrayList = new ArrayList<>();
        for (Integer num : array) {
            arrayList.add(num);
        }
        
        List<Integer> answerList = new ArrayList<>();
        
        for (int i = 0; i < commands.length; i++) {
            int start = commands[i][0] - 1;
            int end = commands[i][1];
            int k = commands[i][2] - 1;
            
            List<Integer> copiedSubList = new ArrayList<>(arrayList.subList(start, end));
            Collections.sort(copiedSubList);
            answerList.add(copiedSubList.get(k));
        }
        
        return answerList;
    }
}